build_scripts/build_usd.py

```diff
@@ -1306,6 +1659,24 @@ def InstallOpenImageIO(context, force, buildArgs):
                      '-DOIIO_BUILD_TESTS=OFF',
                      '-DUSE_PYTHON=OFF',
                      '-DSTOP_ON_WARNING=OFF']
+        if context.targetIos:
```

Comment:
Once again, we could use a different version of OIIO on iOS. Can I request that we upstream fixes to OIIO, as I am sure @lgritz is able to assist integration.

Reply:
I don't have a way to test on iOS myself, but I'm pretty sure that over the years, people have contributed patches to OIIO to fix the iOS build, so I'm presuming that at some point it worked (maybe still?). 

I'm not an iOS developer (nor a MacOS developer per se, except in the trivial sense that I use it all day long but almost entirely pretend it's just a generic Unix toolchain), so I don't know -- is it possible to build on MacOS but cross-compile for iOS, or even emulate iOS? If those tools are available on GitHub Actions runners, I'd be happy to add (that is, accept a PR from somebody who knows what they're doing) another entry to the test matrix for OIIO's CI, wherein we do an emulated iOS build just to ensure that it keeps working and we don't break it with future changes.

```python
if MacOS():
    # OIIO 2.3.15 adds fixes for Apple Silicon cross compilation.
    OIIO_URL = "https://github.com/OpenImageIO/oiio/archive/refs/tags/v2.3.15.0.zip"
else:
    OIIO_URL = "https://github.com/OpenImageIO/oiio/archive/Release-2.1.16.0.zip"
```

Oy, I am definitely available to advise anybody who would like to bring this up to a modern version. The current series is 2.4, and 2.1.16 dates from 2020.

Reply:
Here's an example of github CI for iOS ~ I don't know if it's a useful reference for OIIO's automation ~ but it does work for me :) https://github.com/LabSound/LabSound/blob/main/.github/workflows/cmake-ios.yml

Since OpenImageIO isn't one of the vfxplatform specified versions, I can look into why it's set at 2.1.16.

Reply:
Agree to do it upstream and we can start with the core components!